### ISSUE TITLE
Leelaz of 9 board topped cgos with a score of 3235.

### ISSUE THREAD

**Issue by sbf2000** [leela-zero/leela-zero#1291]
http://www.yss-aya.com/cgos/9x9/cross/CLeela9Test12.html.
One 1080ti GPU was used to generate more than 3.2 million selfplay in more than 3 months.

**Comment by l1t1** [comment]
where download

**Comment by gcp** [comment]
>where download

This :-)

Is this compatible with normal Leela Zero if one just changes ```#define BOARD_SIZE 19```?

**Comment by Alderi-Tokori** [comment]
Nice :)

**Comment by sbf2000** [comment]
_Is it compatible with normal Leela Zero if one just changes #define BOARD_SIZE 19?_
## Yes,totally compatible.

**Comment by sbf2000** [comment]
_What architecture (number of blocks and number of filters) did you use?_
# The weight of the tested is 20block,128filters;The selfplay weight is now 6block,128filter.

**Comment by Alderi-Tokori** [comment]
I see, thank you!

**Comment by Hersmunch** [comment]
So no change to training to support draws with a Komi of 7?

**Comment by Marcin1960** [comment]
where is the download?

**Comment by sbf2000** [comment]
_So no change to training to support draws with a Komi of 7?_
###  Two weights were used,play white according to weight of training with Komi 6.5,and paly black according to weight of training with Komi 7.5.

**Comment by hydrogenpi** [comment]
no bragging rights without published weights

**Comment by sbf2000** [comment]
The weight of training with Komi 6.5 is announced in QQ group.

**Comment by hydrogenpi** [comment]
Announced or published? in any case if published, a link would be best, not everyone has a qq account

**Comment by hydrogenpi** [comment]
and 13 by 13?

**Comment by Marcin1960** [comment]
@sbf2000 "The weight of training with Komi 6.5 is announced in QQ group."

Link, please?

**Comment by gcp** [comment]
>Two weights were used,play white according to weight of training with Komi 6.5,and paly black according to weight of training with Komi 7.5.

Thanks. Any particular reason for doing it this way instead of using komi = 7.0?

**Comment by sbf2000** [comment]
Download Link: 
https://pan.baidu.com/s/1rwndU38j3CZfxj4obGnQoQ
password：d97g

**Comment by sbf2000** [comment]
_Any particular reason for doing it this way instead of using komi = 7.0?_
### 1.komi = 7.0 is rarely used in reality, and komi = 7.5 or komi = 6.5 weights are more useful.
2.komi = 7.5 or komi = 6.5 does not require modification of the source code and is easy to implement.
3. By training the two weights of komi = 7.5 and komi = 6.5, it can be basically proved that the most balanced komi of 9x9 board is between 6.5 and 7.5.

**Comment by sbf2000** [comment]
First:The weight file is for both players.
Second:It's OK.
@gcp

**Comment by gcp** [comment]
Mirrored here:
http://storage.sjeng.org/zero/20-128-6.5.txt.gz

**Comment by billyswong** [comment]
> Two weights were used,play white according to weight of training with Komi 6.5,and paly black according to weight of training with Komi 7.5.

> The weight file is for both players.

So I guess the self-play is set such that while they used the same weight file, white plays as if the game is in komi 6.5, while black plays as if the game is in komi 7.5? Else you saying the weight file work for both sides doesn't make sense to me.

**Comment by Hersmunch** [comment]
@gcp How much effort would it be to add 9x9 releases now that there's some weights available?

**Comment by Ishinoshita** [comment]
@sbf2000 To train two value heads with different komi, I assume you either had each selfplay game to be played till end and so to disable resign feature, or you have modify the resign code to take into account both heads ?

**Comment by sbf2000** [comment]
_So I guess the self-play is set such that while they used the same weight file, white plays as if the game is in komi 6.5, while black plays as if the game is in komi 7.5? Else you saying the weight file work for both sides doesn't make sense to me._

In the course of self-play training,only replace the komi from 7.5 to 6.5,and the resulting weight is the same for both black and white,which is 6.5.
I was just testing on cgos,passing an GTP proxy,using two different weights at the same time(one komi 6.5,another komi 7.5),opening two Leelaz instances,and the proxy passing the  black to an Leelaz,Pass the white to another.

**Comment by KyleBiedermann** [comment]
I noticed the lack of Zen on CGOS hopefully this didn't affect the rating accuracy too much considering the previous versions reached about 3000 range with it playing.

**Comment by KyleBiedermann** [comment]
Yes I noticed I usually work some games from CGOS in my database to compare data, I am just saying that that the rating may be a little inflated due to lack of zen programs as it draws against it quite a few times.

**Comment by Hersmunch** [comment]
It is now top of bayes elo table too http://www.yss-aya.com/cgos/9x9/bayes.html

**Comment by hydrogenpi** [comment]
perhaps an officual 13 by 13 can be started by the project now that experience had been gained with 9 and 19

**Comment by PhilipFRipper** [comment]
@hydrogenpi let us know how it goes

**Comment by diadorak** [comment]
I am still confused. How do I use this? Can I load the network using _leelaz -w 20_128-6.5.txt_?

Does LZ know automatically the board size?

**Comment by alreadydone** [comment]
https://github.com/gcp/leela-zero/issues/1291#issuecomment-385343206
I think you need to change the line and recompile.

**Comment by LetterRip** [comment]
Is the raw data going to be made available as well?

**Comment by sbf2000** [comment]
The raw data is in the precess of sorting and will be released to the net disk later.Most of the old networks have been deleted.
@LetterRip

**Comment by Alderi-Tokori** [comment]
It's beating minus 35 to 5 so it's quite a bit stronger. I've set minus to use these weights on OGS until sunday if anyone wants to have a go at it.

**Comment by rexl2018** [comment]
> The raw data is in the precess of sorting and will be released to the net disk later.Most of the old networks have been deleted.
Waiting for your good news!

**Comment by y-ich** [comment]
Hi.

Please let me advertise my outcome.
I implemented JavaScript version of Pyaq (https://github.com/ymgaq/Pyaq) using WebDNN (https://mil-tokyo.github.io/webdnn/) and have equipped it with the weights here.

Now you can enjoy to play Go with strong AI on your browser!

https://y-ich.github.io/A9.js/index.html
(Sorry, it is Japanese page.)

On my old iMac(2012), the time of one evaluation is about 50ms, not so fast but it is strong enough.

I enjoyed my little project and thank authors of Pyaq(@ymgaq), WebDNN(@Kiikurage, @milhidaka), @sbf2000, and Leela Zero people so much!

**Comment by Tsun-hung** [comment]
LZ has been bugfixed several time since you opened this issue, do you catch up and continue to train the new networks?

**Comment by KyleBiedermann** [comment]
@y-ich cheers on the project of yours it looks good and in english now. I was curious if their was a way in the future to maybe implement custom positions (edit the board) to test various responds the program does in certain situations due to once a player knows the winning lines rarely does the program deviate from the previous game, as I have noticed. Keep up the great work!

**Comment by jokkebk** [comment]
@y-ich great work! Games looked nice, and I lost against it, which is to be expected.

Special thanks for using jGoBoard, nice to see my library in use!

**Comment by ivorget** [comment]
@sbf2000 

> I haven't started training 13x13 yet.

A few months after you said this you had [a very high ranked 13x13 bot on CGOS](http://www.yss-aya.com/cgos/13x13/cross/CLeela13Test7.html). Can we possibly have the weights for that too?
And thanks for the 9x9 weights which work great.

**Comment by trunterzx** [comment]
@sbf2000 

Same request from here. It would help if you could share the weights for the 13x13 board as well.

Also would there be any updates to the 9x9 weights?

**Comment by trunterzx** [comment]
@alreadydone 

Thanks for sharing! Didn't know weights can be found there as well.

Just to check, in the particular comment that you shared the links bring me to download .7z files.
Does it work for the leela zero particularly described in this thread?


The weights shared by **ihavoid** at the top of the other thread works fine, but I wonder how strong is the 9x9 compared to the one posted here.

**Comment by alreadydone** [comment]
You'd need to unzip the 7z file to get a txt file which you can use with leelaz compiled with boardsize 9. Both are trained with 7.5 komi and I think that one is stronger.

**Comment by trunterzx** [comment]
> You'd need to unzip the 7z file to get a txt file which you can use with leelaz compiled with boardsize 9. 

@alreadydone I tried that and I got a very strange situation where the starting move for black only had less than 30%... Is this normal?

**Comment by alreadydone** [comment]
That's normal. It's trained under7.5 komi and the fair komi is probably 7.

**Comment by katagoni** [comment]
> Download Link:
> https://pan.baidu.com/s/1rwndU38j3CZfxj4obGnQoQ
> password：d97g
  

I cant join panbaidu please send me a file google drive or my email
asd739@daum.net xiexie

**Comment by katagoni** [comment]
> _Is it compatible with normal Leela Zero if one just changes #define BOARD_SIZE 19?_
> 
> ## Yes,totally compatible.

How to change 9x9 to 19x19net
